src/org/ods/component/Pipeline.groovy

```diff
@@ -151,8 +151,17 @@ class Pipeline implements Serializable {
 
                         if (!registry.get(NexusService)) {
                             logger.debug 'Registering NexusService'
-                            registry.add(NexusService, new NexusService(
-                                context.nexusUrl, context.nexusUsername, context.nexusPassword))
+                            script.withCredentials([
+                                script.usernamePassword(
+                                    credentialsId: context.credentialsId,
+                                    usernameVariable: 'USERNAME',
+                                    passwordVariable: 'PASSWORD'
+                                )
+                            ]) {
+                                NexusService nexusService = new NexusService(context.nexusUrl,
+                                    script.env.USERNAME as String, script.env.PASSWORD as String)
```

Comment:
It is not a good practice to store clear-text credentials in long-living attributes. It would be better to use withCredentials inside the NexusService whenever needed. Is there anything that prevents us from doing this?